File: test_chat.py
#!/usr/bin/env python3
import socketio
import time
import sys

# Criar cliente socket.io
sio = socketio.Client()
session_id = None
message_count = 0

@sio.event
def connect():
    print("✅ Conectado ao servidor")
    # Criar sessão
    sio.emit('create_session', {'title': 'Teste Automatizado'})

@sio.event
def session_created(data):
    global session_id
    session_id = data.get('sessionId', data.get('id'))
    print(f"📁 Sessão criada: {session_id}")
    
    if session_id:
        # Enviar primeira mensagem
        print('\n📤 Enviando: "o que é claude?"')
        sio.emit('message', {
            'sessionId': session_id,
            'content': 'o que é claude?'
        })

@sio.event
def processing_step(data):
    step = data.get('step', '')
    if step:
        print(f"⚙️  {step}")

@sio.event
def message_stream(data):
    content = data.get('content', '')
    if content and len(content) > 0:
        preview = content[:50] + '...' if len(content) > 50 else content
        print(f"🌊 Streaming: {preview}")

@sio.event
def message_complete(data):
    global message_count
    message_count += 1
    
    content = data.get('content', '')
    print(f"\n✅ Mensagem {message_count} completa!")
    
    if content:
        preview = content[:200] + '...' if len(content) > 200 else content
        print(f"📝 Resposta: {preview}")
    
    if message_count == 1:
        # Enviar segunda mensagem
        time.sleep(2)
        print('\n📤 Enviando: "do que se trata esse projeto?"')
        sio.emit('message', {
            'sessionId': session_id,
            'content': 'do que se trata esse projeto?'
        })
    else:
        # Teste concluído
        print("\n🎉 Teste concluído com sucesso!")
        time.sleep(1)
        sio.disconnect()
        sys.exit(0)

@sio.event
def error(data):
    print(f"❌ Erro: {data}")
    sys.exit(1)

@sio.event
def disconnect():
    print("🔌 Desconectado")

# Conectar ao servidor
try:
    print("🔗 Conectando ao servidor...")
    sio.connect('http://localhost:8080')
    
    # Aguardar até 30 segundos
    time.sleep(30)
    print("⏱️  Timeout - encerrando")
    
except Exception as e:
    print(f"❌ Erro de conexão: {e}")
    sys.exit(1)
finally:
    sio.disconnect()

File: test-chat.js
const io = require('socket.io-client');

// Conectar ao servidor
const socket = io('http://localhost:8080');

let sessionId = null;

socket.on('connect', () => {
  console.log('✅ Conectado ao servidor');
  
  // Criar nova sessão
  socket.emit('create_session', { title: 'Teste Automatizado' });
});

socket.on('session_created', (data) => {
  console.log('📁 Sessão criada:', data.sessionId);
  sessionId = data.sessionId;
  
  // Enviar primeira mensagem
  console.log('\n📤 Enviando primeira mensagem: "o que é claude?"');
  socket.emit('message', {
    sessionId: sessionId,
    content: 'o que é claude?'
  });
});

socket.on('processing_step', (data) => {
  console.log('⚙️ Processando:', data.step, data.details || '');
});

socket.on('message_stream', (data) => {
  if (data.content) {
    console.log('🌊 Streaming:', data.content.substring(0, 50) + '...');
  }
});

socket.on('message_complete', (data) => {
  console.log('\n✅ Resposta completa recebida!');
  console.log('📝 Conteúdo:', data.content.substring(0, 200) + '...');
  console.log('📊 Tokens usados:', data.usage?.output_tokens || 0);
  
  // Após receber primeira resposta, enviar segunda mensagem
  if (data.content.includes('Claude')) {
    setTimeout(() => {
      console.log('\n📤 Enviando segunda mensagem: "do que se trata esse projeto?"');
      socket.emit('message', {
        sessionId: sessionId,
        content: 'do que se trata esse projeto?'
      });
    }, 2000);
  } else {
    // Se já é a segunda resposta, encerrar
    setTimeout(() => {
      console.log('\n✅ Teste concluído com sucesso!');
      process.exit(0);
    }, 2000);
  }
});

socket.on('error', (error) => {
  console.error('❌ Erro recebido:', error);
  process.exit(1);
});

socket.on('disconnect', () => {
  console.log('🔌 Desconectado do servidor');
});

// Timeout de segurança
setTimeout(() => {
  console.log('⏱️ Timeout - encerrando teste');
  process.exit(0);
}, 30000);